CMake/HighFiveTargetDeps.cmake

```diff
@@ -0,0 +1,70 @@
+# Link against target system libs
+# -------------------------------
+
+# therefore making it possible to have new dependencies each build
+if(NOT TARGET highfive_deps)
+  add_library(highfive_deps INTERFACE)
+else()
+# Reset if imported
+set_target_properties(highfive_deps PROPERTIES
+  INTERFACE_COMPILE_DEFINITIONS ""
+  INTERFACE_COMPILE_OPTIONS ""
+  INTERFACE_INCLUDE_DIRECTORIES ""
+  INTERFACE_LINK_LIBRARIES ""
+  INTERFACE_LINK_OPTIONS ""
+  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES ""
+)
+endif()
+
+
+# HDF5
+if(NOT DEFINED HDF5_C_LIBRARIES)
+  set(HDF5_NO_FIND_PACKAGE_CONFIG_FILE TRUE)  # Consistency
+  set(HDF5_PREFER_PARALLEL ${HIGHFIVE_PARALLEL_HDF5})
+  find_package(HDF5 REQUIRED)
+endif()
+
+if(HIGHFIVE_PARALLEL_HDF5 AND NOT HDF5_IS_PARALLEL)
+  message(WARNING "Parallel HDF5 requested but libhdf5 doesnt support it")
+endif()
+
+target_include_directories(highfive_deps SYSTEM INTERFACE ${HDF5_INCLUDE_DIRS})
+target_link_libraries(highfive_deps INTERFACE ${HDF5_C_LIBRARIES})
+target_compile_definitions(highfive_deps INTERFACE ${HDF5_DEFINITIONS})
+
+# Boost
+if(USE_BOOST)
+  set(Boost_NO_BOOST_CMAKE TRUE)  # Consistency
+  find_package(Boost REQUIRED COMPONENTS system serialization)
+  # Dont use imported targets yet, not avail before cmake 3.5
+  target_include_directories(highfive_deps SYSTEM INTERFACE ${Boost_INCLUDE_DIR})
+  target_compile_definitions(highfive_deps INTERFACE BOOST_ALL_NO_LIB)  # Disable windows auto link
+endif()
+
+# MPI
+if(HIGHFIVE_PARALLEL_HDF5 OR HDF5_IS_PARALLEL)
+  find_package(MPI REQUIRED)
+  target_include_directories(highfive_deps SYSTEM INTERFACE ${MPI_CXX_INCLUDE_PATH})
+  target_link_libraries(highfive_deps INTERFACE ${MPI_CXX_LIBRARIES})
+  if(CMAKE_VERSION VERSION_LESS 3.13)
+    target_link_libraries(highfive_deps INTERFACE ${MPI_CXX_LINK_FLAGS})
+  else()
+    target_link_options(highfive_deps INTERFACE "SHELL:${MPI_CXX_LINK_FLAGS}")
+  endif()
+endif()
+
+# Propagate to HighFive
+target_link_libraries(HighFive INTERFACE highfive_deps)
+target_compile_definitions(HighFive INTERFACE MPI_NO_CPPBIND)  # No c++ bindings
+if(USE_BOOST)
+  target_compile_definitions(HighFive INTERFACE H5_USE_BOOST)
+endif()
+
+# Ensure we activate at least C++11
+if(NOT DEFINED CMAKE_CXX_STANDARD)
```

Comment:
I think that `if(NOT DEFINED CMAKE_CXX_STANDARD)` should be only for old CMake. The whole idea that new CMake should be able to deal with different targets having different `target_compile_features`.

Reply:
`target_compile_features` seems the best way of doing it in this case to propagate the flag to all targets that depend on it. However `set(CMAKE_CXX_STANDARD)` is still the most used way of setting the standard, since it is weird to compile different targets with different standards.

Reply:
When using modern CMake one should **never** use `set(CMAKE_CXX_STANDARD)` since it brutally enforces it for **all** builds (imagine you are building more than one executable). Rather `target_compile_features` sandboxes settings such that they are only set for targets build with the set target. I.e. one should **only** touch `set(CMAKE_CXX_STANDARD)` if it is not possible otherwise because you are using an old version of CMake. I find it bad practice to enforce the old brutal way to people who did make the switch.

Reply:
I think I misread the code ☺️
You are in fact not *setting* `CMAKE_CXX_STANDARD`, you are just shielding yourself from overwriting it. In that case: my bad! Never mind, makes sense.